pkg/stub/replset.go

```diff
@@ -132,6 +133,78 @@ func (h *Handler) handleReplsetInit(m *v1alpha1.PerconaServerMongoDB, replset *v
 	return ErrNoRunningMongodContainers
 }
 
+func (h *Handler) handleStatefulSetUpdate(m *v1alpha1.PerconaServerMongoDB, set *appsv1.StatefulSet, replset *v1alpha1.ReplsetSpec, resources *corev1.ResourceRequirements) error {
+	var doUpdate bool
+
+	// Ensure the stateful set size is the same as the spec
+	err := h.client.Get(set)
+	if err != nil {
+		return fmt.Errorf("failed to get stateful set for replset %s: %v", replset.Name, err)
+	}
+	if *set.Spec.Replicas != replset.Size {
+		logrus.Infof("setting replicas to %d for replset: %s", replset.Size, replset.Name)
+		set.Spec.Replicas = &replset.Size
+		doUpdate = true
+	}
+
+	// Ensure the PSMDB version is the same as the spec
+	expectedImage := getPSMDBDockerImageName(m)
+	mongod := &set.Spec.Template.Spec.Containers[0]
+	if mongod != nil && mongod.Image != expectedImage {
```

Comment:
no need in check `mongod != nil`. At previous line either we got `runtime error: index out of range` or took the address of some `struct` which isn't `nil-able` type.